src/main/scala/zio/s3/Live.scala

```diff
@@ -210,8 +214,12 @@ final class Live(unsafeClient: S3AsyncClient) extends S3.Service {
                   )
     } yield ()
 
-  def execute[T](f: S3AsyncClient => CompletableFuture[T]): ZIO[Any, S3Exception, T] =
+  override def execute[T](f: S3AsyncClient => CompletableFuture[T]): ZIO[Any, S3Exception, T] =
     ZIO.fromCompletionStage(f(unsafeClient)).refineToOrDie[S3Exception]
+
+  override def executePublisher[T](f: S3AsyncClient => SdkPublisher[T]): ZStream[Any, S3Exception, T] =
```

Comment:
can you explain why to expose this ?
What is the use case ?

Reply:
i would not expose this method

Reply:
My idea of exposing the `executePublisher` is the same as `execute`: the methods that are present in `S3AsyncClient`, but absent in zio-s3, can be used via wrapping the value into ZStream/ZIO by calling `execute*`.
Now there are 3 methods returning the publisher present: `listMultipartUploadsPaginator`, `listObjectsV2Paginator` and `listObjectVersionsPaginator`. I can technically expose them and hide the `executePublisher`. However, in this case, any additional methods on `S3AsyncClient` wouldn't be exposed in any way to the zio-s3 user in case they are added to the `S3AsyncClient`. 

`listAllObjects` was added just to replace the custom implementation `listObjectsDescendant` and as an example usage of the `executePublisher`. The current `listObjectsDescendant` cannot be used in an application that uses ZLayer, because the `listObjectsDescendant` is not present on the layer level. So
```
ZLayer.fromService[S3.Service, MyService]{ s3 => ... }
```
in this situation `s3.listObjectsDescendant` is not present. And I need it. So I added the `listAllObjects` which has the same implementation as `listObjectsDescendant` but is provided by the aws sdk rather than being rewritten in zio-s3.

To sum up this message: I would add the streaming methods that I need (`listMultipartUploadsPaginator` and `listObjectsV2Paginator`) explicitly into the `S3.Service`. And I would also keep the `executePublisher` in order to cover any other methods from aws sdk that can potentially appear in newer versions of sdk.

Reply:
Oki then the solution is to move all the methods `streamLines(), paginate(), listObjectsDescendant()` inside 
`trait S3.Service`

Reply:
Concerning exposing the publisher, most of the methods in AsyncClient return A `CompletableFuture` and the other methods like `listObjectsV2Paginator` are already implemented or have an equivalent with future.